sdk/trace/tracer.go

```diff
@@ -17,10 +23,34 @@ type tracer struct {
 
 	provider             *TracerProvider
 	instrumentationScope instrumentation.Scope
+
+	selfObservabilityEnabled bool
+	spanLiveMetric           otelconv.SDKSpanLive
+	spanStartedMetric        otelconv.SDKSpanEnded // TODO (#7027): Replace with otelconv.SDKSpanStarted when it is available.
 }
 
 var _ trace.Tracer = &tracer{}
 
+func (tr *tracer) initSelfObservability(mp metric.MeterProvider) {
```

Comment:
nit: I would remove the MeterProvider argument from this function to simplify it

```golang
func (tr *tracer) initSelfObservability() {
	if !strings.ToLower(os.Getenv("OTEL_GO_X_SELF_OBSERVABILITY")) == "true" {
		return
	}
	tr.selfObservabilityEnabled = true
	m := otel.GetMeterProvider().Meter("go.opentelemetry.io/otel/sdk/trace", ...
```

This means we don't need the nil check on meterProvider, and consolidates logic for initializing self-obs to this function.